lldb-eval/parser.cc

```diff
@@ -2893,15 +2893,19 @@ ExprResult Parser::BuildMemberOf(ExprResult lhs, std::string member_id,
   if (is_arrow) {
     // "member of pointer" operator, check that LHS is a pointer and
     // dereference it.
-    if (!lhs_type.IsPointerType()) {
+    if (!lhs_type.IsPointerType() && !lhs_type.IsArrayType()) {
       BailOut(ErrorCode::kInvalidOperandType,
               llvm::formatv("member reference type '{0}' is not a pointer; did "
                             "you mean to use '.'?",
                             lhs_type.GetName()),
               location);
       return std::make_unique<ErrorNode>();
     }
-    lhs_type = lhs_type.GetPointeeType();
+    if (lhs_type.IsPointerType()) {
+      lhs_type = lhs_type.GetPointeeType();
+    } else {
+      lhs_type = lhs_type.GetArrayElementType();
```

Comment:
If `lhs` is an array, we should insert an `array-to-pointer` conversion here. This way the interpreter doesn't need to do anything.

Reply:
Done